File: split/charwarning.pas
procedure charwarning(f: internalfontnumber; c: eightbits);
begin
  if eqtb[5298].int > 0 then
  begin
    begindiagnostic;
    printnl(824);
    print(c);
    print(825);
    slowprint(fontname[f]);
    printchar(33);
    enddiagnostic(false);
  end;
end;

File: cpp/charwarning.cpp
#include "charwarning.h"
#include "printchar.h"
#include "print.h"
#include "begindiagnostic.h"
#include "printnl.h"
#include "slowprint.h"
#include "printchar.h"
#include "enddiagnostic.h"

void charwarning(internalfontnumber f, eightbits c)
{
	if (tracing_lost_chars() > 0)
	{
		begindiagnostic();
		printnl("Missing character: There is no ");
		printchar(c);
		print(" in font ");
		slowprint(fontname[f]);
		printchar('!');
		enddiagnostic(false);
	}
}
